cdap-security-spi/src/main/java/io/cdap/cdap/security/spi/authorization/RoleController.java

```diff
@@ -0,0 +1,78 @@
+package io.cdap.cdap.security.spi.authorization;
+
+import io.cdap.cdap.api.security.AccessException;
+import io.cdap.cdap.proto.security.GrantedPermission;
+import io.cdap.cdap.proto.security.Principal;
+import io.cdap.cdap.proto.security.Role;
+
+import java.util.Set;
+
+/**
+ * This interface is responsible to manage role related actions and it's relation with {@link Principal}
+ *
+ * This logic was previously used in {@link AccessController}
+ * This is an API .
```

Comment:
Please don't add API to SPI module.

Reply:
The GrantFetcher and PermissionManager are also right now in SPI module. 
So I placed here..

Reply:
I know, but those were primarily SPI, so we can't move those unfortunately.